### PR TITLE
[FluentFileInput] adds a "disabled" property to the FluentFileInput component

### PR DESCRIPTION
**Pull request by NickHirras**
# Pull Request

## 📖 Description

I'm using the FluentFileInput component in a project. I have a requirement to _disable_ the file upload field under certain conditions. 

All HTML input components, including the native `input type=file`, support a disabled attribute. 

This pull request adds a `disabled` property to the `FluentFileInput` component.  When this attribute is set, a CSS class is applied to the container making it appear more translucent / lighter, as well as disabling cursor events on the component, and setting the embedded `input type=file` component to be disabled. 

This PR also includes updated unit tests to cover the new property.

This PR also updates the demo project to add an example of this component in the disabled state.

image

## 👩‍💻 Reviewer Notes

I considered trying to add all the other missing standard input component attributes, but I wasn't comfortable with that level of change. And it's outside the scope of what I need at the moment.  It may be more appropriate to extend input component instead of base component and inherit those additional attributes.

### General

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fluentui-blazor/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have added [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for my new compontent
- [x] I have modified an existing component
- [x] I have validate [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for an existing component

### CODE REVIEW

**Comment by NickHirras** [comment]
@microsoft-github-policy-service agree

File: tests/Core/Microsoft.FluentUI.AspNetCore.Components.Tests.csproj [hunk 1]
```diff
@@ -59,6 +59,10 @@
       <ParentFile>$([System.String]::Copy('%(FileName)').Split('.')[0])</ParentFile>
       <DependentUpon>%(ParentFile).razor</DependentUpon>
     </None>
+    <None Update="InputFile\FluentInputFileTests.FluentInputFile_Disabled.verified.razor.html">
+      <ParentFile>FluentInputFileTests</ParentFile>
+      <DependentUpon>FluentInputFileTests.razor</DependentUpon>
+    </None>
```
[the remark this answers is not in the document]
Comment (NickHirras) [reply tests/Core/Microsoft.FluentUI.AspNetCore.Components.Tests.csproj]: This refernce has been removed.

File: examples/Demo/Shared/Pages/InputFile/Examples/InputFileDisabled.razor
[the remark this answers is not in the document]
Comment (NickHirras) [reply examples/Demo/Shared/Pages/InputFile/Examples/InputFileDisabled.razor:1]: The demo has been simplified, along with the ID made unique.

**Comment by NickHirras** [comment]
@vnbaaij requested changes have been pushed.

**Review by vnbaaij — APPROVED** [review]

File: src/Core/Components/InputFile/FluentInputFile.razor.css [hunk 2]
```diff
@@ -35,4 +35,10 @@
         text-align: center;
         line-height: 8px;
         font-size: var(--type-ramp-minus-2-font-size);
-    }
\ No newline at end of file
+    }
+
+    .fluent-inputfile-container.disabled {
```
[the remark this answers is not in the document]
Comment (vnbaaij) [reply src/Core/Components/InputFile/FluentInputFile.razor.css:40]: Fixing it in separate commit (also for v3)

**Pull request merged**